test/engine/PreconditionActionTest.cpp

```diff
@@ -0,0 +1,48 @@
+//   Copyright 2025, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include <gmock/gmock.h>
+
+#include "../util/IndexTestHelpers.h"
+#include "./ValuesForTesting.h"
+#include "engine/PreconditionAction.h"
+#include "engine/QueryExecutionTree.h"
+
+// _____________________________________________________________________________
+TEST(PreconditionAction, basicFunctionality) {
+  EXPECT_EQ(
+      PreconditionAction{PreconditionAction::IMPLICITLY_SATISFIED}.getTree(),
+      std::nullopt);
+  EXPECT_EQ(
+      PreconditionAction{PreconditionAction::SATISFY_EXTERNALLY}.getTree(),
+      std::nullopt);
```

Comment:
You definitely want global static variables of type `PreconditionAction` s.t. you can write
`Expect_eq(PreconditiionAction::Implicitly_Satisfied}.getTree(), std::nullopt)`
(the tags can almost be an  implementation detail, much less boilerplate.

Reply:
(Some occurences in tests + allows the constructors to be explicit.)

Reply:
Okay, I see that they are, so the outer constructor is just redundant.